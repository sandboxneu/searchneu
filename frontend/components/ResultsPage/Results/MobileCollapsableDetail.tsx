import React from 'react'
import IconCollapseExpand from '../../images/IconCollapseExpand'

interface MobileCollapsableDetailProps {
  title: string
  expand: boolean
  setExpand: (b: boolean) => void
  renderChildren: () => (JSX.Element | any[])
}

function MobileCollapsableDetail({
  title, expand, setExpand, renderChildren,
}: MobileCollapsableDetailProps) {
  return (
    <div className='MobileSearchResult__panel--collapsableContainer' role='button' tabIndex={ 0 } onClick={ () => setExpand(!expand) }>
      <div className='MobileSearchResult__panel--collapsableTitle'>
        <IconCollapseExpand width='6' height='12' fill='#000000' className={ expand ? 'MobileSearchResult__panel--rotatedIcon' : '' } />
        <span>{title}</span>
      </div>
      {expand && <div>{renderChildren()}</div>}
    </div>
  )
}

export default MobileCollapsableDetail
