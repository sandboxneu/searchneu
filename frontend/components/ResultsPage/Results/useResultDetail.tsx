import React from 'react'
import { useHistory } from 'react-router-dom'
import macros from '../../macros'
import RequisiteBranch from '../../classModels/RequisiteBranch'
import Course from '../../classModels/Course';


export default function useResultDetail(aClass: Course) {
  const history = useHistory();
  const onReqClick = (reqType, childBranch, event, hash) => {
    history.push(hash);

    // Create the React element and add it to retVal
    const searchEvent = new CustomEvent(macros.searchEvent, { detail: `${childBranch.subject} ${childBranch.classId}` });
    window.dispatchEvent(searchEvent);
    event.preventDefault();

    // Rest of this function is analytics
    const classCode = `${childBranch.subject} ${childBranch.classId}`;
    let reqTypeString;

    switch (reqType) {
      case macros.prereqTypes.PREREQ:
        reqTypeString = 'Prerequisite';
        break;
      case macros.prereqTypes.COREQ:
        reqTypeString = 'Corequisite';
        break;
      case macros.prereqTypes.PREREQ_FOR:
        reqTypeString = 'Required Prerequisite For';
        break;
      case macros.prereqTypes.OPT_PREREQ_FOR:
        reqTypeString = 'Optional Prerequisite For';
        break;
      default:
        macros.error('unknown type.', reqType);
    }

    macros.logAmplitudeEvent('Requisite Click', {
      type: reqTypeString,
      subject: childBranch.subject,
      classId: childBranch.classId,
      classCode: classCode,
    });
  }

  // returns an array made to be rendered by react to display the prereqs
  const getReqsString = (reqType, course) => {
    const retVal = [];

    // Keep track of which subject+classId combonations have been used so far.
    // If you encounter the same subject+classId combo in the same loop, skip the second one.
    // This is because there is no need to show (eg. CS 2500 and CS 2500 (hon)) in the same group
    // because only the subject and the classId are going to be shown.
    const processedSubjectClassIds = {};

    let childNodes;

    if (reqType === macros.prereqTypes.PREREQ) {
      childNodes = course.prereqs;
    } else if (reqType === macros.prereqTypes.COREQ) {
      childNodes = course.coreqs;
    } else if (reqType === macros.prereqTypes.PREREQ_FOR) {
      childNodes = course.prereqsFor;
    } else if (reqType === macros.prereqTypes.OPT_PREREQ_FOR) {
      childNodes = course.optPrereqsFor;
    } else {
      macros.error('Invalid prereqType', reqType);
    }

    childNodes.values.forEach((childBranch) => {
      // If the childBranch is a class
      if (!(childBranch instanceof RequisiteBranch)) {
        if (childBranch.isString) {
          // Skip if already seen
          if (processedSubjectClassIds[childBranch.desc]) {
            return;
          }
          processedSubjectClassIds[childBranch.desc] = true;
          retVal.push(childBranch.desc);
        } else {
          // Skip if already seen
          if (processedSubjectClassIds[childBranch.subject + childBranch.classId]) {
            return;
          }
          processedSubjectClassIds[childBranch.subject + childBranch.classId] = true;

          // When adding support for right click-> open in new tab, we might also be able to fix the jsx-a11y/anchor-is-valid errors.
          // They are disabled for now.
          const hash = `/${aClass.termId}/${childBranch.subject}${childBranch.classId}`;

          const element = (
            <a
              role='link'
              key={ hash }
              tabIndex={ 0 }
              onClick={ (event) => { onReqClick(reqType, childBranch, event, hash); } }
            >
              {`${childBranch.subject} ${childBranch.classId}`}
            </a>
          );

          retVal.push(element);
        }
      } else if (reqType === macros.prereqTypes.PREREQ) {
        // Figure out how many unique classIds there are in the prereqs.
        const allClassIds = {};
        for (const node of childBranch.prereqs.values) {
          allClassIds[node.classId] = true;
        }

        // If there is only 1 prereq with a unique classId, don't show the parens.
        if (Object.keys(allClassIds).length === 1) {
          retVal.push(getReqsString(macros.prereqTypes.PREREQ, childBranch));
        } else {
          retVal.push(['(', getReqsString(macros.prereqTypes.PREREQ, childBranch), ')']);
        }
      } else {
        macros.error('Branch found and parsing coreqs?', childBranch);
      }
    })


    // Now insert the type divider ("and" vs "or") between the elements.
    // If we're parsing prereqsFor, we should use just a comma as a separator.
    // Can't use the join in case the objects are react elements
    if (reqType === macros.prereqTypes.PREREQ_FOR || reqType === macros.prereqTypes.OPT_PREREQ_FOR) {
      for (let i = retVal.length - 1; i >= 1; i--) {
        retVal.splice(i, 0, ', ');
      }
    } else {
      let type;
      if (reqType === macros.prereqTypes.PREREQ) {
        type = aClass.prereqs.type;
      } else if (reqType === macros.prereqTypes.COREQ) {
        type = aClass.coreqs.type;
      }
      for (let i = retVal.length - 1; i >= 1; i--) {
        retVal.splice(i, 0, ` ${type} `);
      }
    }

    if (retVal.length === 0) {
      return (
        <span className='empty'>
          None
        </span>
      );
    }

    return retVal;
  }

  const optionalDisplay = (prereqType, course) => {
    const data = getReqsString(prereqType, course);

    return data;
  }

  const creditsString = () => {
    const creditDescriptor = aClass.maxCredits > 1 || aClass.maxCredits === 0 ? 'CREDITS' : 'CREDIT'
    return aClass.maxCredits === aClass.minCredits ? `${aClass.maxCredits} ${creditDescriptor}` : `${aClass.maxCredits}-${aClass.maxCredits} ${creditDescriptor}`
  }


  return {
    optionalDisplay,
    creditsString,
  }
}
