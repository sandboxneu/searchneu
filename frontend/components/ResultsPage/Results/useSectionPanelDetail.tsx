import Section from '../../classModels/Section';


interface UseSectionPanelDetailReturn {
  getSeatsClass: () => string;
}


export default function useSectionPanelDetail(section: Section): UseSectionPanelDetailReturn {
  const getSeatsClass = () => {
    const seatingPercentage = section.seatsRemaining / section.seatsCapacity
    if (seatingPercentage > (2 / 3)) {
      return 'green'
    } if (seatingPercentage > (1 / 3)) {
      return 'yellow'
    }
    return 'red'
  }

  return {
    getSeatsClass: getSeatsClass,
  }
}
