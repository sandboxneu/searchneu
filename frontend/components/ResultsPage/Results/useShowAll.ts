import { useState, useEffect } from 'react'
import Course from '../../classModels/Course'
import Section from '../../classModels/Section'

interface UseShowAllReturn {
  showAll: boolean
  setShowAll: (b: boolean) => void
  renderedSections: Section[]
  hideShowAll: boolean
}

export default function useShowAll(aClass: Course) : UseShowAllReturn {
  const [showAll, setShowAll] = useState(false)

  const sectionsShownByDefault = aClass.sections.length < 3 ? aClass.sections.length : 3
  const [renderedSections, setRenderedSections] = useState(aClass.sections.slice(0, sectionsShownByDefault))
  const hideShowAll = sectionsShownByDefault === aClass.sections.length

  useEffect(() => {
    if (showAll) {
      setRenderedSections(aClass.sections)
    } else {
      setRenderedSections(aClass.sections.slice(0, sectionsShownByDefault))
    }
  }, [aClass.sections, sectionsShownByDefault, showAll])

  return {
    showAll: showAll,
    setShowAll: setShowAll,
    renderedSections: renderedSections,
    hideShowAll: hideShowAll,
  }
}
