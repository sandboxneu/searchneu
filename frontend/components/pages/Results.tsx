/*
 * This file is part of Search NEU and licensed under AGPL3.
 * See the license file in the root folder for details.
 */
import React, { useCallback } from 'react';
import _ from 'lodash';
import { useHistory, useParams } from 'react-router-dom';
import {
  useQueryParams, BooleanParam, useQueryParam,
} from 'use-query-params';
import useDeepCompareEffect from 'use-deep-compare-effect';
import logo from '../images/logo.svg';
import search from '../search';
import macros from '../macros';
import SearchBar from '../ResultsPage/SearchBar';
import Footer from '../Footer';
import useSearch from '../ResultsPage/useSearch';
import FilterPanel from '../ResultsPage/FilterPanel';
import FilterPills from '../ResultsPage/FilterPills';
import FeedbackModal from '../ResultsPage/FeedbackModal/FeedbackModal';
import EmptyResultsContainer from '../ResultsPage/EmptyResultsContainer';
import MobileSearchOverlay from '../ResultsPage/MobileSearchOverlay';
import useAtTop from '../ResultsPage/useAtTop';
import {
  FilterSelection, QUERY_PARAM_ENCODERS, DEFAULT_FILTER_SELECTION, areFiltersSet,
} from '../ResultsPage/filters';
import ResultsLoader from '../ResultsPage/ResultsLoader';
import { BLANK_SEARCH_RESULT, SearchResult } from '../types';
import { termDropdownOptions, campusDropdownOptions } from '../types';
import SearchDropdown from '../ResultsPage/SearchDropdown';



interface SearchParams {
  campus: string,
  termId: string,
  query: string,
  filters: FilterSelection
}

let count = 0;
// Log search queries to amplitude on enter.
function logSearch(searchQuery: string) {
  searchQuery = searchQuery.trim();

  if (searchQuery) {
    count++;
    macros.logAmplitudeEvent('Search', { query: searchQuery.toLowerCase(), sessionCount: count });
  }
}

// Retreive result data from backend.
const fetchResults = async ({ query, termId, filters }: SearchParams, page: number): Promise<SearchResult> => {
  const response: SearchResult = await search.search(query, termId, filters, (1 + page) * 10);
  if (page === 0) {
    logSearch(query);
  }
  return response;
};

export default function Results() {
  const atTop = useAtTop();
  const [showOverlay, setShowOverlay] = useQueryParam('overlay', BooleanParam);
  const { campus, termId, query = '' } = useParams();
  const [qParams, setQParams] = useQueryParams(QUERY_PARAM_ENCODERS);
  const history = useHistory();

  const setSearchQuery = (q: string) => { history.push(`/${campus}/${termId}/${q}${history.location.search}`); }
  const setTerm = (t: string) => { history.push(`/${campus}/${t}/${query}${history.location.search}`); }
  const setCampus = (c: string) => { history.push(`/${c}/${termId}/${query}${history.location.search}`); }

  const filters: FilterSelection = _.merge({}, DEFAULT_FILTER_SELECTION, qParams);

  const searchParams: SearchParams = { campus, termId, query, filters };

  const filtersAreSet: Boolean = areFiltersSet(filters);

  const us = useSearch(searchParams, BLANK_SEARCH_RESULT(), fetchResults);
  const {
    isReady, loadMore, doSearch,
  } = us;
  const { results, filterOptions } = us.results;

  useDeepCompareEffect(() => {
    doSearch(searchParams);
  }, [searchParams, doSearch]);

  if (showOverlay && macros.isMobile) {
    return (
      <MobileSearchOverlay
        query={ query }
        filterSelection={ filters }
        filterOptions={ filterOptions }
        setFilterPills={ setQParams }
        setQuery={ (q: string) => setSearchQuery(q) }
        onExecute={ () => setShowOverlay(false) }
      />
    )
  }

  return (
    <div>
      <div className={ `Results_Header ${atTop ? 'Results_Header-top' : ''}` }>
        <img src={ logo } className='Results__Logo' alt='logo' onClick={ () => { history.push('/'); } } />
        <div className='Results__spacer' />
        <div className='Results__searchwrapper'>
          <SearchBar
            onSearch={ setSearchQuery }
            query={ query }
            onClick={ () => {
              if (macros.isMobile) {
                setShowOverlay(true);
              }
            } }
          />
        </div>
        <div className="Breadcrumb_Container">
          <div className="Breadcrumb_Container__dropDownContainer">
            <SearchDropdown
              options ={ campusDropdownOptions }
              value={ campus }
              placeholder="NEU"
              onChange={ setCampus }
              className="searchDropdown"
              compact={ false }
            />
          </div>
          <span className="Breadcrumb_Container__slash">/</span>
          <div className="Breadcrumb_Container__dropDownContainer">
            <SearchDropdown
              options={ termDropdownOptions }
              value={ termId }
              placeholder="Fall 2020"
              onChange={ setTerm }
              className="searchDropdown"
              compact={ false }
            />
          </div>
        </div>
      </div>
      {!macros.isMobile && <FeedbackModal />}
      <div className='Results_Container'>
        {!macros.isMobile && (
          <>
            <div className='Results_SidebarWrapper'>
              <FilterPanel
                options={ filterOptions }
                selected={ filters }
                setActive={ setQParams }
              />
            </div>
            <div className='Results_SidebarSpacer' />
          </>
        )}
        <div className='Results_Main'>
          { filtersAreSet
          && <FilterPills filters={ filters } setFilters={ setQParams } />}
          {!isReady && <div style={{ visibility: 'hidden' }} />}
          {isReady && results.length === 0 && <EmptyResultsContainer query={ query } filtersAreSet={ filtersAreSet } setFilters={ setQParams } /> }
          {isReady && results.length > 0
            && (
              <ResultsLoader
                results={ results }
                loadMore={ loadMore }
              />
            )}
          <Footer />
        </div>
      </div>
      <div className='botttomPadding' />
    </div>

  );
}
