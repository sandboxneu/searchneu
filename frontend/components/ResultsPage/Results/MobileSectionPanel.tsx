import React from 'react'
import Section from '../../classModels/Section'
import WeekdayBoxes from './WeekdayBoxes'
import NotifCheckBox from '../../panels/NotifCheckBox'
import IconGlobe from '../../images/IconGlobe'
import useSectionPanelDetail from './useSectionPanelDetail';
import { MomentTuple } from '../../classModels/Meeting';

interface MobileSectionPanelProps {
  section: Section
  showNotificationSwitches: boolean
}

function MobileSectionPanel({ section, showNotificationSwitches } : MobileSectionPanelProps) {
  const { getSeatsClass } = useSectionPanelDetail(section)

  const groupedTimesAndDays = (times: MomentTuple[]) => {
    const daysOfWeek = ['Su', 'M', 'T', 'W', 'Th', 'F', 'S']
    return times.reduce((acc, t) => {
      const timeString = `${t.start.format('h:mm')}-${t.end.format('h:mm a')}`
      acc.set(timeString, acc.get(timeString) ? acc.get(timeString) + daysOfWeek[t.start.day()] : daysOfWeek[t.start.day()])

      return acc
    }, new Map())
  }

  const getMeetings = (s: Section) => {
    return s.meetings.map((m) => (
      Array.from(groupedTimesAndDays(m.times)).map(([time, days]) => (
        <>
          <span className='MobileSectionPanel__meetings--time'>
            {`${days}, ${time} | ${m.location}`}
          </span>
          <br />
        </>
      ))
    ))
  }


  return (
    <div className='MobileSectionPanel'>
      <div className='MobileSectionPanel__header'>
        <span>{section.getProfs().join(', ')}</span>
        <span>Boston</span>
      </div>
      <div className='MobileSectionPanel__firstRow'>
        <div>
          <a
            target='_blank'
            rel='noopener noreferrer'
            href={ section.url }
          >
            <IconGlobe />
          </a>
          <span>{section.crn}</span>
        </div>
        {showNotificationSwitches && <NotifCheckBox section={ section } />}
      </div>
      <div className='MobileSectionPanel__secondRow'>
        {!section.online && <WeekdayBoxes meetingDays={ section.getDaysOfWeekAsBooleans() } />}
      </div>
      <div className='MobileSectionPanel__meetings'>
        {section.online ? <span className='MobileSectionPanel__meetings--online'>Online Class</span>
          : getMeetings(section)}
      </div>
      <div className={ getSeatsClass() }>
        {`${section.seatsRemaining}/${section.seatsCapacity} Seats Available `}
      </div>
    </div>
  )
}

export default MobileSectionPanel
