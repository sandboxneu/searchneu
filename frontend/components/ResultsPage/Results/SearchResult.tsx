import React from 'react'
import macros from '../../macros'
import DesktopSectionPanel from './DesktopSectionPanel'
import Course from '../../classModels/Course'
import IconGlobe from '../../images/IconGlobe'
import IconArrow from '../../images/IconArrow'
import SignUpForNotifications from '../../SignUpForNotifications'
import useResultDetail from './useResultDetail'
import useUserChange from './useUserChange';
import useShowAll from './useShowAll';

interface SearchResultProps {
  aClass: Course,
}
export default function SearchResult({ aClass } : SearchResultProps) {
  const { optionalDisplay, creditsString } = useResultDetail(aClass)
  const userIsWatchingClass = useUserChange(aClass)
  const {
    showAll, setShowAll, renderedSections, hideShowAll,
  } = useShowAll(aClass)

  const feeString = aClass.feeDescription && aClass.feeAmount ? `${aClass.feeDescription}- $${aClass.feeAmount}` : null

  return (
    <div className='SearchResult'>
      <div className='SearchResult__header'>
        <div className='SearchResult__header--left'>
          <span className='SearchResult__header--classTitle'>
            {aClass.subject} {aClass.classId}: {aClass.name}
          </span>
          <div className='SearchResult__header--sub'>
            <a
              target='_blank'
              rel='noopener noreferrer'
              data-tip={ `View on ${aClass.host}` }
              href={ aClass.prettyUrl }
            >
              <IconGlobe />
            </a>
            <span>{`Updated ${(aClass.getLastUpdateString())}`}</span>
          </div>
        </div>
        <span className='SearchResult__header--creditString'>
          {creditsString()}
        </span>
      </div>
      <div className='SearchResult__panel'>
        {aClass.desc}
        <br />
        <br />
        <div className='SearchResult__panel--main'>
          <div className='SearchResult__panel--left'>
            NUPaths:
            {aClass.nupath.length > 0 ? <span> {aClass.nupath.join(', ')}</span> : <span className='empty'> None</span>}
            <br />
            Prerequisites: {optionalDisplay(macros.prereqTypes.PREREQ, aClass)}
            <br />
            Corequisites: {optionalDisplay(macros.prereqTypes.COREQ, aClass)}
            <br />
            Course fees:
            {feeString ? <span>  {feeString}</span> : <span className='empty'> None</span>}
          </div>
          <div className='SearchResult__panel--right'>
            <SignUpForNotifications aClass={ aClass } userIsWatchingClass={ userIsWatchingClass } />
          </div>
        </div>
      </div>
      <table className='SearchResult__sectionTable'>
        <thead>
          <tr>
            <th>
              <div className='inlineBlock' data-tip='Course Reference Number'>
                CRN
              </div>
            </th>
            <th> Professors </th>
            <th> Meetings </th>
            <th> Campus </th>
            <th> Seats </th>
            {userIsWatchingClass && <th> Notifications </th>}
          </tr>
        </thead>
        <tbody>
          {renderedSections.map((section) => (
            <DesktopSectionPanel
              key={ section.crn }
              section={ section }
              showNotificationSwitches={ userIsWatchingClass }
            />
          ))}
        </tbody>
      </table>
      {!hideShowAll
      && (
      <div className='SearchResult__showAll' role='button' tabIndex={ 0 } onClick={ () => setShowAll(!showAll) }>
        <span>{showAll ? 'Collapse sections' : 'Show all sections'}</span>
        <IconArrow className={ showAll ? 'SearchResult__showAll--collapse' : null } />
      </div>
      )}
    </div>
  )
}
