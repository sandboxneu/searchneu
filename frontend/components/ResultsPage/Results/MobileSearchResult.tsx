import React, { useState } from 'react'
import Course from '../../classModels/Course'
import IconCollapseExpand from '../../images/IconCollapseExpand'
import IconArrow from '../../images/IconArrow'
import useResultDetail from './useResultDetail'
import useUserChange from './useUserChange'
import useShowAll from './useShowAll'
import macros from '../../macros'
import MobileCollapsableDetail from './MobileCollapsableDetail'
import MobileSectionPanel from './MobileSectionPanel'
import SignUpForNotifications from '../../SignUpForNotifications'


interface MobileSearchResultProps {
  aClass: Course,
}

function MobileSearchResult({ aClass } : MobileSearchResultProps) {
  const [expanded, setExpanded] = useState(false)
  const [showMore, setShowMore] = useState(false)
  const [showNUPath, setShowNUPath] = useState(false)
  const [showPrereq, setShowPrereq] = useState(false)
  const [showCoreq, setShowCoreq] = useState(false)
  const userIsWatchingClass = useUserChange(aClass)
  const {
    showAll, setShowAll, renderedSections, hideShowAll,
  } = useShowAll(aClass)

  const { optionalDisplay, creditsString } = useResultDetail(aClass);


  const renderNUPaths = () => (
    // eslint-disable-next-line react/prop-types
    <div>{aClass.nupath.length > 0 ? <div> {aClass.nupath.join(', ')}</div> : <span className='empty'> None</span>}</div>
  )

  return (
    <div className='MobileSearchResult'>
      <div className={ expanded ? 'MobileSearchResult__header--expanded' : 'MobileSearchResult__header' } role='button' tabIndex={ 0 } onClick={ () => setExpanded(!expanded) }>
        <IconCollapseExpand />
        <span className='MobileSearchResult__header--classTitle'>
          {`${aClass.subject} ${aClass.classId} : ${aClass.name}`}
        </span>
      </div>
      {expanded && (
      <div className='MobileSearchResult__panel'>
        <div className='MobileSearchResult__panel--mainContainer'>
          <div className='MobileSearchResult__panel--infoStrings'>
            <a href={ aClass.prettyUrl } target='_blank' rel='noopener noreferrer'>{`Updated ${(aClass.getLastUpdateString())}`}</a>
            <span>
              {creditsString()}
            </span>
          </div>
          <div className={ showMore ? 'MobileSearchResult__panel--description' : 'MobileSearchResult__panel--descriptionHidden' }>
            {aClass.desc}
          </div>
          <div className='MobileSearchResult__panel--showMore' role='button' tabIndex={ 0 } onClick={ () => setShowMore(!showMore) }>{showMore ? 'Show less' : 'Show more'}</div>
          <MobileCollapsableDetail title='NUPATH' expand={ showNUPath } setExpand={ setShowNUPath } renderChildren={ renderNUPaths } />
          <MobileCollapsableDetail title='PREREQUISITES' expand={ showPrereq } setExpand={ setShowPrereq } renderChildren={ () => optionalDisplay(macros.prereqTypes.PREREQ, aClass) } />
          <MobileCollapsableDetail title='COREQUISITES' expand={ showCoreq } setExpand={ setShowCoreq } renderChildren={ () => optionalDisplay(macros.prereqTypes.COREQ, aClass) } />
          <div className='MobileSearchResult__panel--notifContainer'>
            <SignUpForNotifications aClass={ aClass } userIsWatchingClass={ userIsWatchingClass } />
          </div>
        </div>
        <div className='MobileSearchResult__panel--sections'>
          {
            renderedSections.map((section) => (
              <MobileSectionPanel
                key={ section.crn }
                section={ section }
                showNotificationSwitches={ userIsWatchingClass }
              />
            ))
          }
        </div>
        {!hideShowAll && (
        <div className='MobileSearchResult__showAll' role='button' tabIndex={ 0 } onClick={ () => setShowAll(!showAll) }>
          <span>{showAll ? 'Collapse sections' : 'Show all sections'}</span>
          <IconArrow className={ showAll ? 'MobileSearchResult__showAll--collapse' : '' } />
        </div>
        )}
      </div>
      )}
    </div>

  )
}

export default MobileSearchResult
