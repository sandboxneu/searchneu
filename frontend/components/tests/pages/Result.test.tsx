/*
 * This file is part of Search NEU and licensed under AGPL3.
 * See the license file in the root folder for details.
 */

import React from 'react';

import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Results from '../../pages/Results';

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: jest.fn(),
  }),
  useParams: () => ({ campus: 'NEU', termId:'202030', query: 'cs' }),
}));

Enzyme.configure({ adapter: new Adapter() });

it('should render a section', () => {
  const result = shallow(<Results />);
  expect(result).toMatchSnapshot();
});
