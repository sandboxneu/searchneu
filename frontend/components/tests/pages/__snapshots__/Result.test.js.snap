// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a section 1`] = `
<Fragment>
  <div
    className="Results_Header"
  >
    <img
      alt="logo"
      className="Results__Logo"
      onClick={[Function]}
      src="file"
    />
    <div
      className="Results__spacer"
    />
    <div
      className="Results__searchwrapper"
    >
      <SearchBar
        onSearch={[Function]}
        query="cs"
      />
    </div>
    <Memo(TermDropdown)
      compact={true}
      onChange={[Function]}
      termId="202030"
    />
  </div>
  <div
    className="Results_Container"
  >
    <div>
      <div
        style={
          Object {
            "visibility": "hidden",
          }
        }
      >
        <div
          className="subjectContaineRowContainer"
        />
        <ResultsLoader
          loadMore={[Function]}
          results={Array []}
        />
      </div>
    </div>
    <div
      className="botttomPadding"
    />
    <Memo(Footer) />
  </div>
</Fragment>
`;
